tests/conftest.py

```diff
@@ -86,6 +88,13 @@ def pytest_unconfigure(config):
     if not config.getoption("--no-teardown"):
         stop_docker_compose()
 
+def pytest_xdist_node_collection_finished(node, ids):
```

Comment:
Is this some special function that gets called with xdist?

If so, it won't work without xdist, because the lockfile will be used even if xdist isn't present. Isn't there some pytest initialization function we can move this to? One that executes before xdist starts forking processes.